File: problems/2078/2078-1.cpp
#include <vector>

class Solution {
public:
    int maxDistance(std::vector<int>& colors) {
        int res = 0;

        int n = colors.size() - 1;
        int i = n;
        while (colors[0] == colors[i]) {
            --i;
        }
        res = std::max(res, i);

        i = 0;
        while (colors[i] == colors[n]) {
            ++i;
        }
        res = std::max(res, n - i);

        return res;
    }
};

File: problems/2078/2078-1.py
class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        res = 0

        n = len(colors) - 1
        i = n
        while colors[0] == colors[i]:
            i -= 1
        res = max(res, i)

        i = 0
        while colors[n] == colors[i]:
            i += 1
        res = max(res, n - i)

        return res
